dyninstAPI/src/linux.C

```diff
@@ -174,118 +180,167 @@ bool PCEventMuxer::useCallback(Dyninst::ProcControlAPI::EventType et)
    return false;
 }
 
-bool BinaryEdit::getResolvedLibraryPath(const string &filename, std::vector<string> &paths) {
-    char *libPathStr, *libPath;
-    std::vector<string> libPaths;
-    struct stat dummy;
-    char buffer[512];
-    char *pos, *key, *val;
+namespace
+{
+template <typename ContainerT = std::vector<std::string>,
+          typename PredicateT = std::string (*)(const std::string&)>
+inline ContainerT
+delimit(
```

Comment:
For as much as I hate in/out parameters, I'd rather replace this with 

    std::vector<std::string> vals;
    boost::split(vals, line, boost::token_compress_on);

just to have one less hand-rolled solution in this codebase.

Reply:
Makes perfect sense. I've actually had this fix for a while but hadn't submitted it bc I figured that would be the case and hadn't looked around for any existing solution.